crates/sui-types/src/gas.rs

```diff
@@ -321,8 +321,8 @@ pub fn check_gas_balance(gas_object: &Object, gas_budget: u64) -> SuiResult {
 
     let balance = get_gas_balance(gas_object)?;
     ok_or_gas_error!(
-        balance >= gas_budget,
-        format!("Gas balance is {balance}, not enough to pay {gas_budget}")
+        (balance as u128) >= (gas_budget as u128) * (gas_price as u128),
+        format!("Gas balance is {balance}, not enough to pay {gas_budget} units with unit price of {gas_price}")
```

Comment:
It might be clearer to show the total amount required instead of showing gas_budget and price separately.

Reply:
was thinking to convey the actual gas_price that is used in calculation in case we have difference between storage_price and computation_price. Do you think it makes sense to have all of the like 
`not enough to pay total budget amount of {total_amount} with unit price of {price}`?

Reply:
Doesn't hurt to have more numbers!